src/hash.rs

```diff
@@ -13,21 +14,29 @@ pub struct Hash(
 
 impl Hash {
     /// Creates a new Hash from last hash of `n` consecutive hashes of an item.
-    pub fn from_chain<T: AsRef<[u8]>>(value: &T, n: usize) -> Self {
-        let mut hash = Self::from(value);
+    /// Adds in the prefix bytes for each hash.
+    pub fn from_chain(prefix: impl AsRef<[u8]>, value: impl AsRef<[u8]>, n: usize) -> Self {
+        let mut hash = Self::from(&prefix, value);
         for _ in 1..n {
-            hash = Self::from(&hash);
+            hash = Self::from(&prefix, hash);
         }
 
         hash
     }
 
-    pub fn from<T: AsRef<[u8]>>(value: &T) -> Self {
+    /// Creates a keyed Hash by hashing the value with given prefix.
+    pub fn from(prefix: impl AsRef<[u8]>, value: impl AsRef<[u8]>) -> Self {
         let mut hasher = Sha3_256::new();
+        hasher.update(prefix.as_ref());
         hasher.update(value.as_ref());
         Self(hasher.finalize().into())
     }
 
+    /// Creates a random Hash.
+    pub fn random(rng: &mut impl RngCore) -> Self {
```

Comment:
ditto - https://github.com/wnfs-wg/rs-skip-ratchet/pull/29#discussion_r1147259110

Reply:
Yeah, this was actually unused, so I removed it.